ouroboros-consensus-cardano-tools/src/Cardano/Tools/DBSynthesizer/Orphans.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Tools.DBSynthesizer.Orphans () where
+
+import           Cardano.Node.Types (AdjustFilePaths (..),
+                     NodeByronProtocolConfiguration (..),
+                     NodeHardForkProtocolConfiguration (..))
+import           Cardano.Tools.DBSynthesizer.Types
+
+import qualified Cardano.Chain.Update as Byron (ApplicationName (..))
+import           Cardano.Crypto (RequiresNetworkMagic (..))
+
+import           Data.Aeson as Aeson (FromJSON (..), withObject, (.!=), (.:),
+                     (.:?))
+
+import           Control.Monad (when)
+
+
+instance FromJSON NodeConfigStub where
+    parseJSON val = withObject "NodeConfigStub" (parse' val) val
+      where
+        parse' o v = do
+          proto <- v .: "Protocol"
+          when (proto /= ("Cardano" :: String)) $
+            fail $ "nodeConfig.Protocol expected: Cardano; found: " ++ proto
+          NodeConfigStub o
+            <$> v .: "AlonzoGenesisFile"
+            <*> v .: "ShelleyGenesisFile"
+            <*> v .: "ByronGenesisFile"
+
+instance AdjustFilePaths NodeConfigStub where
+    adjustFilePaths f nc@NodeConfigStub{..} =
+        nc {
+            ncsAlonzoGenesisFile    = f ncsAlonzoGenesisFile
+          , ncsShelleyGenesisFile   = f ncsShelleyGenesisFile
+          , ncsByronGenesisFile     = f ncsByronGenesisFile
+          }
+
+instance AdjustFilePaths NodeCredentials where
+    adjustFilePaths f nc@NodeCredentials{..} =
+        nc {
+            credCertFile  = f <$> credCertFile
+          , credVRFFile   = f <$> credVRFFile
+          , credKESFile   = f <$> credKESFile
+          , credBulkFile  = f <$> credBulkFile
+          }
+
+-- DUPLICATE: mirroring parsers from cardano-node/src/Cardano/Node/Configuration/POM.hs
```

Comment:
Silly question maybe, but why can't we use these?

Reply:
The constraint of this new project was for consensus not to depend on anything in the node repository. So those parts which are relevant for setting up block forging had to be ported to consensus.